sdk/src/main/scala/com/horizen/api/http/SidechainTransactionApiRoute.scala

```diff
@@ -447,6 +446,107 @@ case class SidechainTransactionApiRoute(override val settings: RESTApiSettings,
+    }
+  }
+
+  private def buildOpenStakeTransactionSimplified(body: ReqOpenStakeSimplified): Try[OpenStakeTransaction] = {
+    applyOnNodeView { sidechainNodeView =>
+      val wallet = sidechainNodeView.getNodeWallet
+
+      //Collect input box
+      val inputBox = wallet.allBoxes().asScala
+        .filter(box => box.proposition().equals(PublicKey25519PropositionSerializer.getSerializer.parseBytes(BytesUtils.fromHexString(body.forgerProposition))))
+      if (inputBox.isEmpty)
+        throw new IllegalArgumentException("Unable to find input for Proposition "+body.forgerProposition)
+      createAndSignOpenStakeTransaction(wallet, inputBox.head, body.forgerProposition, body.forgerListIndex, body.fee)
+    }
+  }
+
+  /**
+   * Create and sign a openStakeTransaction
+   * @return a signed OpenStakeTransaction or a GenericTransactionError
+   */
+  private def buildOpenStakeTransaction(body: ReqOpenStake): Try[OpenStakeTransaction] = {
+    applyOnNodeView { sidechainNodeView =>
+      val wallet = sidechainNodeView.getNodeWallet
+
+      //Collect input box
+      val inputBox = wallet.allBoxes().asScala
+        .filter(box => BytesUtils.toHexString(box.id()).equals(body.transactionInput.boxId))
+      if (inputBox.isEmpty) {
+        throw new IllegalArgumentException("Unable to find input!")
+      }
+      if (inputBox.length > 1) {
+        throw new IllegalArgumentException("To much inputs found!")
+      }
+
+      createAndSignOpenStakeTransaction(wallet, inputBox.head, body.regularOutputProposition, body.forgerListIndex, body.fee)
+    }
+  }
+
+
+  private def createAndSignOpenStakeTransaction(wallet: NodeWallet, inputBox: Box[Proposition], outputProposition: String,
+                                                forgerListIndex: Int, inputFee: Option[Long]): Try[OpenStakeTransaction] = {
+    //Collect fee
+    val fee = inputFee.getOrElse(0L)
+    if (fee < 0) {
+      throw new IllegalArgumentException("Fee can't be negative!")
+    }
+    if (fee > inputBox.value) {
+      throw new IllegalArgumentException("Fee can't be greater than the input!")
+    }
+
+    //Collect output box
+    var output:JOptional[ZenBoxData] = JOptional.empty()
+    if (fee < inputBox.value()) {
+      output = JOptional.of(new ZenBoxData(
+        PublicKey25519PropositionSerializer.getSerializer.parseBytes(BytesUtils.fromHexString(outputProposition)),
+        inputBox.value() - fee))
+    }
+
+    Try {
+      // Create unsigned tx
+      val unsignedTransaction = new OpenStakeTransaction(inputBox.id(), output, null, forgerListIndex, fee, OpenStakeTransaction.OPEN_STAKE_TRANSACTION_VERSION)
+
+      // Create signed tx.
+      val messageToSign = unsignedTransaction.messageToSign()
+      val proof = wallet.secretByPublicKey(inputBox.proposition()).get().sign(messageToSign).asInstanceOf[Proof[Proposition]]
+      new OpenStakeTransaction(inputBox.id(), output, proof, forgerListIndex, fee, OpenStakeTransaction.OPEN_STAKE_TRANSACTION_VERSION)
```

Comment:
Why not to use `OpenStakeTransaction.create` method? so some parts of the code above can be removed.

Reply:
Good point, thanks!